### PR TITLE
Per-channel rate monitor

### PR DESCRIPTION
**Pull request by florian-toennies**
implemented data rate monitor (and nothing else):
* added:
  * two svg files (monitor_history and monitor_tpc_layout)
  * one stylesheet file (monitor.css)
* replaced
  * script file /javascript/monitor_script.js with monitor_tpc_datarate.js
* modified:
  * route monitor.js (added many new routes to download relevant database entries)
  * monitor.jade redid the entire view

### CODE REVIEW

File: routes/monitor.js [hunk 1]
```diff
@@ -12,24 +12,321 @@ router.get('/', ensureAuthenticated, function(req, res) {
...
+        }
+    );
+    
 });
 
+
+
+// returns cable_map (used to generate new tpc svg)
+// no authentintification as python will need to access these functions
+router.get('/get_cable_map', function(req,res){
```
Comment (darrylmasson) [inline routes/monitor.js]: No unauthenticated routes. If you need to regenerate the svg then run the script from somewhere with database access like the gateway (or old-gw)

File: routes/monitor.js [hunk 2]
```diff
@@ -12,24 +12,321 @@ router.get('/', ensureAuthenticated, function(req, res) {
...
+    var collection = runs_db.get('cable_map');
+    
+    collection.find({},{},function(e,docs){
+        res.json(docs);
+    })
+});
+
+// returns board layout to update svg
+// no authentintification as python will need to access these functions
+router.get('/get_options_map', function(req,res){
```
Comment (darrylmasson) [inline routes/monitor.js]: Here as well

File: routes/monitor.js [hunk 3]
```diff
@@ -12,24 +12,321 @@ router.get('/', ensureAuthenticated, function(req, res) {
...
-	    return res.send(JSON.stringify(collInfos));
+    console.log(db)
+    
+    var collection = db.get("status");
+    var mongo_pipeline = [];
+    
+    // only take reader 0 to 2 and non empty channel entries
+    mongo_pipeline.push({
+        "$match":{
+            "host": {$regex:"reader[0-2]_reader_0", $options:"$i" },
```
Comment (darrylmasson) [inline routes/monitor.js]: We will probably be spinning up reader3 soon, so we should tweak the regex to accept it.

File: routes/monitor.js [hunk 4]
```diff
@@ -12,24 +12,321 @@ router.get('/', ensureAuthenticated, function(req, res) {
...
-    var q = url.parse(req.url, true).query;
-    var coll = q.coll;
+    var collection = db.get('status');
+        ObjectID = req.ObjectID;
+    
+    var oid_min = ObjectID(0);
+    
+    // start mongo pipeline with empty array
+    var mongo_pipeline = []
+    mongo_pipeline.push({"$match":{"host":{$regex:"reader[0-2]_reader_0", $options:"$i" }}});
```
Comment (darrylmasson) [inline routes/monitor.js]: Also here

**Comment by florian-toennies** [comment]
* removed the unsecured routes
* modified regex to also accept reader3 (was in there three times (also in line 199))

**Comment by darrylmasson** [comment]
Given the debacle that was the test of #10, this might need rebasing on to master to work.

**Comment by darrylmasson** [comment]
The DAQ isn't currently running so "live" results aren't available right now, but it looks pretty good so I'll probably merge it. I'll fix the few style issues myself, but it would be very nice to have a view overlay of the VME crates. Also, in the reader view, three channels from the top array aren't listed correctly (the most recent cable map fixes this issue), and all the high-energy channels are missing.

**Comment by AlexElykov** [comment]
Regarding layout. Do we want the deadtime monitor on the same web page just beow this one, or on a different one?

**Comment by darrylmasson** [comment]
> @darrylmasson Regarding layout. Do we want the deadtime monitor on the same web page just below this rate monitor, or on a different one?

No, that would do better on the Status page

**Comment by florian-toennies** [comment]
I started working on a new version of the svg creator python script (it now access the database directly, using the same environmental variable as nodiaq). A few questions regarding this:
1. The high-energy channels are the ones with channel number >= 1000, right?
2. The channels from 500 to 999 are the just the amplified normal channels and should get ignored?

as we now do not have a continuous range of channel numbers, the javascript part needs a little bit more rework...

**Comment by darrylmasson** [comment]
High-energy channels are 500-752, channels >= 1000 are from the muon veto. See https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:dsg:daq:channel_groups for the full set.
Channels 500-752 should be displayed with anything per-digitizer (so the reader display, or the VME crate display)

**Pull request merged**